app/src/test/contrib/fetch/DemographicFetchTests.ts

```diff
@@ -38,54 +38,44 @@ describe("DemographicStore.fetchOneTimeToken", () => {
         },
         makePayload: () => "TOKEN",
-        expectedURL: "/touchstones/touchstoneId/demographics/source2/typeId/get_onetime_link/?gender=female"
+        expectedURL: "/touchstones/touchstoneId/demographics/sourceId/typeId/get_onetime_link/?gender=female"
     });
 
     helper.addTestsToMocha();
 
     it("does not include gender parameter where it is not applicable", (done: DoneCallback) => {
         const touchstone = mockTouchstone({ id: "touchstoneId" });
-        const dataSet = mockDemographicStatisticType({ id: "typeId", gender_is_applicable: false });
+        const dataSet = mockDemographicDataset({ id: "typeId", gender_is_applicable: false });
         touchstoneActions.setCurrentTouchstone(touchstone.id);
         demographicActions.update([dataSet]);
         demographicActions.selectDataSet(dataSet.id);
-        demographicActions.selectSource("sourceId");
 
         const fetcherSpy = helper.getFetcherSpy();
         demographicStore.fetchOneTimeToken().catch(doNothing);
         checkAsync(done, () => {
             expect(fetcherSpy.args).to.have.length(1, "Fetch method was not invoked");
-            expect(fetcherSpy.args[0][0]).to.equal("/touchstones/touchstoneId/demographics/sourceId/typeId/get_onetime_link/");
+            expect(fetcherSpy.args[0][0]).to.equal("/touchstones/touchstoneId/demographics/source1/typeId/get_onetime_link/");
         });
     });
 
-    it("does not fetch until all required options are selected", () => {
+    it("does not fetch until dataset is selected", () => {
         const stub = helper.sandbox.stubFetch(demographicStore, "_fetchOneTimeToken");
         const touchstone = mockTouchstone({ id: "touchstoneId" });
-        const dataSet = mockDemographicStatisticType({
-            sources: ["source1", "source2"],
+        const dataSet = mockDemographicDataset({
+            sources: ["source1" ],
             id: "typeId"
         });
 
-        // First, show that it does nothing if no data set is selected
+        // show that it does nothing if no data set is selected
         alt.recycle();
         touchstoneActions.setCurrentTouchstone(touchstone.id);
         demographicActions.update([dataSet]);
         demographicStore.fetchOneTimeToken().catch(doNothing);
         expect(stub.called).to.equal(false, "Fetched token even though data set was not selected");
 
-        // Next, show that it does nothing if no source is selected
-        demographicActions.selectDataSet(dataSet.id);
-        demographicStore.fetchOneTimeToken().catch(doNothing);
-        expect(stub.called).to.equal(false, "Fetched token even though data set was not selected");
 
-        // Next, select a source as well and show that this makes the difference
-        demographicActions.selectSource(dataSet.sources[0]);
-        demographicStore.fetchOneTimeToken().catch(doNothing);
-        expect(stub.called).to.equal(true, "Did not fetch token even though data set was selected");
```

Comment:
Previously this test showed that it didn't fetch until it was meant to, and then that it did fetch once you had selected all the options.

Now it just shows that it doesn't fetch if you don't select a data set. Shouldn't it also check that it does fetch when you have selected one?